File: TP1/2_palindromo_C/Palindromo.c
/**
 * Marcos Antonio Lommez Candido Ribeiro
 * 77157
 * AED's 2
 * 10/08/2022
 */

#include <stdio.h>
#include <stdbool.h>
#include <string.h>

/**
 * Metodo que recebe uma String e testa se
 * a mesma e' um palindromo, exemplo:
 * arara, reviver, ama, ovo
 *
 * @param s String a ser testada
 * @return True se for um palindromo
 */
bool ePalindromo(char s[])
{
    if (strlen(s) == 0)
    {
        return false;
    }

    for (int i = 0, j = (strlen(s) - 1); i < j; i++, j--)
    {
        if (s[i] != s[j])
        {
            return false;
        }
    }
    return true;
}

/**
 * Metodo para testar se a String contem a palavra 'FIM'
 *
 * @param s String a testar
 * @return True se for 'FIM'
 */
bool isFim(char s[])
{
    return (strlen(s) == 3 && s[0] == 'F' && s[1] == 'I' && s[2] == 'M');
}


// Driver
int main()
{
    char entrada[1000];

    for (; !isFim(entrada);)
    {
        scanf(" %1000[^\n]", entrada);
        if (!isFim(entrada))
            printf("%s\n", ePalindromo(entrada) ? "SIM" : "NAO");
    }

    return 0;
}

File: TP1/1_palindromo_Java/Palindromo.java
/**
 * Marcos Antonio Lommez Candido Ribeiro
 * 77157
 * AED's 2
 * 10/08/2022
 */

public class Palindromo {

	/**
	 * Metodo para testar se a String contem a palavra 'FIM'
	 * 
	 * @param s String a testar
	 * @return True se for 'FIM'
	 */
	public static boolean isFim(String s) {
		return (s.length() == 3 && s.charAt(0) == 'F' && s.charAt(1) == 'I' && s.charAt(2) == 'M');
	}

	/**
	 * Metodo que recebe uma String e testa se
	 * a mesma e' um palindromo, exemplo:
	 * arara, reviver, ama, ovo
	 * 
	 * @param s String a ser testada
	 * @return True se for um palindromo
	 */
	public static boolean ePalindromo(String s) {
		for (int i = 0, j = (s.length() - 1); i <= j; i++, j--) {
			if (s.charAt(i) != s.charAt(j)) {
				return false;
			}
		}
		return true;
	}

	// Driver
	public static void main(String[] args) {
		String[] entrada = new String[1000];
		int numEntrada = 0;

		// Leitura da entrada padrao
		do {
			entrada[numEntrada] = MyIO.readLine();
		} while (isFim(entrada[numEntrada++]) == false);
		numEntrada--; // Desconsiderar ultima linha contendo a palavra FIM

		for (int i = 0; i < numEntrada; i++) {
			System.out.println(ePalindromo(entrada[i]) ? "SIM" : "NAO");
		}
	}
}
